### ISSUE TITLE
Refling does not seem to work

### ISSUE THREAD

**Issue by on3iro** [kristijanhusak/orgmode.nvim#145]
### Describe the bug

Hi,

I have just been toying around with orgmode to get a bit familar with it. Unfortunately I was not able to refile anything to a custom location. Refiling to the default file works however.
I have no prior experience with orgmode so maybe I am doing something wrong (it is relatively hard to follow the gifs do get an understanding of the workflow).
So here's what I did:

### Steps to reproduce

1. I captured a todo with `<leader>oc` and `<C-c>`
2. the file now shows up correctly inside my agenda
3. Now I enter the file (by either pressing enter or tab) and put my cursor on the todo item
4. i press `<leader>or` and the destination prompt opens
5. It was unclear to me if the destination path is relative to my orgmode directory, my current working directory or has to be specified exactly, so I tried the following (default orgmode dir is `~/orgmode/org`
    a. work.org (both while my pwd was `~` and `~/orgmode/org`)
    b. ~/orgmode/org/work.org  
6. After typing in the destination I press enter
7. Then nothing happened (my todo remained inside the refile and no new file was created, I also tried to create the exact file beforehand, but it remained empty)

### Expected behavior

The todo item should be moved from the refile into the destination file

### Minimal init.lua

I tried the minimal lua, however I was not able to open the agenda view at all so I could not really test it.

### OS / Distro

OSX 11.6

### Neovim version/commit

0.5.1

**Comment by levouh** [comment]
If you press `<Tab>` you will see the completion results, which are the short names (no extension) of `.org` files within your configured "domain". Select one of those then press `<Tab>` again and you'll then see headlines for that file. Select a headline, and the item will be refilled underneath it.

Does that help? I'm on my phone right now but I'd be happy to post a screen cast if need be as well.

**Comment by on3iro** [comment]
@levouh ah yeah that seemed to have worked at least up until the second `<Tab>` press. That did not do anything. But pressing enter appended the item (without me being able to pick a headline to refile it underneath).

**Comment by on3iro** [comment]
ah ok, now it worked (not sure if I did anything differently) thanks for the help :)

**Comment by levouh** [comment]
No worries, if there are parts that aren't documented well or not super intuitive that should definitely be improved and we can use this issue to track that, otherwise glad you got it figured out.

**Comment by on3iro** [comment]
@levouh There are two more things where I am not quite sure if I am doing something wrong or if these are bugs (or my neovim config does something unintended):

1. I can't seem to change folds. Everything is always unfolded. When pressing `<Tab>` I see a message at the bottom of neovim, indicating that I am indeed changing folding modes, but it does not have any visual effect.
2. I am not able to toggle the state of checkboxes. (on a side note: is it intended that I can't add another checkbox item automatically in insert mode? I know  that I can add one in normal mode by pressing `<leader><CR>`)

Thanks again :)

**Comment by kristijanhusak** [comment]
@on3iro 

1. What is your `&foldmethod` and `&foldexpr` while in org file?
2. Ctrl space toggles checkboxes. All mappings are currently available only in normal mode.

**Comment by on3iro** [comment]
> @on3iro
> 
>     1. What is your `&foldmethod` and `&foldexpr` while in org file?
> 
>     2. Ctrl space toggles checkboxes. All mappings are currently available only in normal mode.

@kristijanhusak
1. My `foldmethod=expr`, `foldlevel=1`, `foldexpr=org#fold_expr()`
2. That was what I was trying, but nothing happenes when I press `<C-space>` in normal mode hovering over a list item with checkbox

One other strange behavior I get is, that sometimes while I type inside an org file, the cursor slips one character (usually when some kind of complete suggestion opens up). So for example, when I type: `something` and a completion opens up after I typed the `e` my cursor will be in front of the `e` like this `som|e`. Ofcourse when I continue typing this will completely mess up the word. I use cocs omni completion as source for org files. Any idea what could cause this?

**Comment by kristijanhusak** [comment]
Uninstall vim-orgmode plugin, it's not compatible with orgmode.nvim, and give everything a second try.

**Comment by on3iro** [comment]
> Uninstall vim-orgmode plugin, it's not compatible with orgmode.nvim, and give everything a second try.

I am sorry I am not sure I understand. I do not have `vim-orgmode` installed - where did you get that assumption from? @kristijanhusak The only installed orgmode plugin is `orgmode.nvim`.

**Comment by kristijanhusak** [comment]
@on3iro fold method that you get is not from orgmode.nvim, something overrides it. Do you use vim-polyglot?

**Comment by on3iro** [comment]
> @on3iro fold method that you get is not from orgmode.nvim, something overrides it. Do you use vim-polyglot?

ahhh yes, i do

**Comment by kristijanhusak** [comment]
Disable org filetype in polyglot, you can find in their readme how to do that.

**Comment by on3iro** [comment]
@kristijanhusak thanks that fixed the folding issue. However checking checkboxes still won't work.

**Comment by kristijanhusak** [comment]
What do you get for `:map <c-space>` command?

**Comment by on3iro** [comment]
`<C-Space>   *@<Cmd>lua require("orgmode").action("org_mappings.toggle_checkbox")<CR>`

**Comment by kristijanhusak** [comment]
@on3iro can you give me an example content that doesn't work for you? I cannot reproduce it.

**Comment by on3iro** [comment]
Ok, I have no idea what happened... but now it suddenly works o.O. I'll keep an eye on it and if the issue turns up again I'll report back. Thanks for helping out @kristijanhusak :)
